test/check_imports.lean

```diff
@@ -129,9 +135,23 @@ def checkStdDataImports : MetaM Unit := do
         checkStdDataDir (autofix := autofix) modMap entry
     let stdImports ← expectedStdImports
     let .some stdMod := modMap.find? `Std
-        | warn "Missing Std module!"
+        | warn false "Missing Std module!; Run `lake build`."
     let warned ← checkMissingImports `Std stdMod stdImports
     if autofix && warned then
       writeImportModule "Std.lean" stdImports
+    match ← getWarningInfo with
```

Comment:
Honestly this looks overcomplicated, all of the below and more are expressed by `diff`.

Reply:
You may be correct on the complicated nature.  The one non-trivial thing to do from shell is avoiding overwriting definitions, e.g., here's output when one module has spurious definitions:

```
/repos/std4 lake build && ./scripts/updateStd.sh
[192/203] Building Std.Data.Array
[255/256] Building Std
Expected Std.Data.Array to not contain additional declarations.
Declarations should be moved out.
This error cannot be automatically fixed.
Found errors that cannot be automatically fixed.
Address unfixable issues and rerun lake build && ./scripts/updateStd.sh.
test/check_imports.lean:157:0: error: Fatal error
```

Here's the usual missing imports:
```
~/repos/std4 lake build && ./scripts/updateStd.sh
[255/256] Building Std
Missing import Std.Data.Array.Lemmas in Std.Data.Array
Generating Std/Data/Array.lean
Missing import Std.CodeAction.Deprecated in Std
Missing import Std.CodeAction.Basic in Std
Generating Std.lean
Found missing imports and attempted fixes.
Run lake build && ./scripts/updateStd.sh to verify.
Multiple runs may be needed.
~/repos/std4 lake build && ./scripts/updateStd.sh
[224/237] Building Std.Data.Array
[255/256] Building Std
```

This is partially an exercise for me to use Lean for file system manipulation to see what should be better.

Reply:
Another thing I wanted to mention but maybe not pursue in this PR is to consider moving this script to a script with a `main` function instead of (ab)using compile-time elaboration execution which necessitates the strange argument passing. That would make it a bit easier to run directly as a script, so we could eliminate the `updateStd.sh` wrapper. (I'm imagining that we will have many scripts like this one eventually, and I would like to be able to write them in lean and run them without additional declarations in the lakefile.)

Reply:
This is really nice code! I think it may be useful to generalize a bit so that it can be reused by other packages. (Obviously, this is not required for this PR.)

Reply:
Thanks, I will merge as is.  I like the idea of making this standalone eventually.

The next checker of a similar nature I have planned is one that looks for missing `simp` lemmas (e.g., checking operations that return a list and are meant for specifications have `simp` lemmas for `List.length` and other core introspection primitives).